yaksh/views.py

```diff
@@ -616,6 +617,54 @@ def show_question(request, question, paper, error_message=None,
     module = course.learning_module.get(id=module_id)
     all_modules = course.get_learning_modules()
     all_question_types = [types[0] for types in question_types]
+    types = {}
+    categorized_questions = {}
+    categorized_questions["objectives"] = questions.filter(Q(type="mcq") |
+                                                           Q(type="mcc") |
+                                                           Q(type="arrange"))
+    categorized_questions["blanks"] = questions.filter(Q(type="integer") |
+                                                       Q(type="string") |
+                                                       Q(type="float"))
+    categorized_questions["programming"] = questions.filter(Q(type="code"))
+    categorized_questions["upload"] = questions.filter(Q(type="upload"))
+
+    types["objective_types"] = set([_type.type
+                                    for _type in
+                                    categorized_questions["objectives"]])
+    types["blank_types"] = set([_type.type
+                                for _type in
+                                categorized_questions["blanks"]])
+    types["programming_types"] = set([_type.type
+                                      for _type in
+                                      categorized_questions["programming"]])
+    types["upload_types"] = set([_type.type
```

Comment:
No need of doing this everytime!
Simply create a constants that contains the objective types and programming types.
Also the above filters are then not needed.
This will be good as everytime a new type is added, then we don't need to modify this code, simply add to the constant!